packages/expo-doctor/src/utils/files.ts

```diff
@@ -2,27 +2,37 @@ import spawnAsync from '@expo/spawn-async';
 import fs from 'fs';
 import path from 'path';
 
+import { isFileIgnoredByRules } from './isFileIgnoredByRules';
+
+const EASIGNORE_FILENAME = '.easignore';
+const GITIGNORE_FILENAME = '.gitignore';
+
 /**
- * Checks if a file exists and is not ignored by git.
+ * Checks if a file exists and is not ignored by EAS (.easignore) or git (.gitignore).
  * @param filePath The path to the file to check.
- * @returns `true` if the file exists and is not ignored by git.
+ * @returns `true` if the file exists and is not ignored.
  */
 export async function existsAndIsNotIgnoredAsync(filePath: string): Promise<boolean> {
   return fs.existsSync(filePath) && !(await isFileIgnoredAsync(filePath));
 }
 
 /**
- * Checks if a file is ignored by git.
+ * Checks if a file is ignored by EAS (.easignore) or git (.gitignore).
+ * Prioritizes .easignore if it exists, otherwise falls back to .gitignore.
  * @param filePath The path to the file to check.
- * @returns `true` if the file is ignored by git.
+ * @returns `true` if the file is ignored, `false` otherwise.
  */
 export async function isFileIgnoredAsync(filePath: string): Promise<boolean> {
```

Comment:
I'm wondering if this function (and the other exported function in this file) should have a flag indicating whether to consider easignore, since there could be tests where we don't want to? Then, the test could indicate if it wants to cater the advice to EAS usage (this could be as simple as checking for eas.json, like `StoreCompatibilityCheck` does).

Reply:
I don't believe this is necessary. `StoreCompatibilityCheck` seems to work fine. 

Here's the check working properly with an old project: 
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/bf92b8b1-05b9-4b7f-a2c0-94add8186cdc">

Here's working fine with a recent project:
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/1cbc65ff-b278-45f4-8dc3-fbfa5c4f3623">

And all tests are passing. Is there a specific case you want me to check?